mr_project/mr_app/views.py

```diff
@@ -1,9 +1,231 @@
+import json
+
 from django.shortcuts import render
 
 # Create your views here.
+from time import time
+from django.http import HttpResponse, HttpResponseBadRequest, HttpResponseNotFound, JsonResponse
+import neo4j
+from django.views.decorators.csrf import csrf_exempt
+from requests import codes
+from . import DRIVER, EMOTIONS, USERS
+from datetime import datetime
+
+#WRITE TRANSACTION FUNCTIONS
+def create_user_tx(tx, name):
+    result = tx.run('CREATE (u:User{name:$name}) RETURN u.name AS username', name=name)
+    record = result.single()
+    result.consume()
+    return record['username']
+
+
+def create_emotion_tx(tx, name):
+    result = tx.run('CREATE (em:Emotion {name:$name})  RETURN em.name AS emotion_label', name=name)
+    record = result.single()
+    result.consume()
+    return record['emotion_label']
+
+
+def create_epoch_relationship_between_user_and_emotion(tx, emotion, user, epoch_timestamp):
+    result = tx.run(f'MATCH (em:Emotion {{name:"{emotion}"}}), (u:User {{name:"{user}"}}) CREATE (u)-[ts:Timestamp{{epoch:"{epoch_timestamp}"}}]->(em) RETURN ts.epoch as seconds')
+    record = result.single()
+    result.consume()
+    return record['seconds']
+
+
+#EFFECTIVE READ TRANSACTIONS TO CHECK IF NODES EXIST OR NOT
+def check_for_user(driver_arg, user_name):
+    with driver_arg.session() as session:
+        result = session.run("MATCH (n:User{ name:$user_name }) RETURN n.name AS name", user_name=user_name)
+        values = result.values()
+        if not values:
+            return False
+        else:
+            return True
+
+
+def check_for_emotion(driver_arg, emotion_name):
+    with driver_arg.session() as session:
+        result = session.run("MATCH (n:Emotion{ name:$emotion_name }) RETURN n.name AS name", emotion_name=emotion_name)
+        values = result.values()
+        if not values:
+            return False
+        else:
+            return True
```

Comment:
This can be simplified by creating a generic helper method
```python
def get_node(driver, label, property, property_value):
  with driver.session() as session:
    return session.run(f'MATCH (node:{label}{{ {property}:{property_value} }}) RETURN node.{property} as {property}').values()

def check_for_user(driver, username):
  return True if get_node(driver, 'User', 'name', username) else False

def check_for_emotion(driver, emotion_name):
  return True if get_node(driver, 'Emotion', 'name', emotion_name) else False
```

Reply:
check